### PR TITLE
PYTHON-609: Align Text and Ascii columns behaviour

### PR DESCRIPTION
**Pull request by kdeldycke**
This PR:
* Add full unit-tests of CQLengine's `Ascii` columns.
* Extend `Text` columns unit-tests for minimal and maximal length.
* Thoroughly cover edge-cases on empty / blank / `None` value strings handling, especially in relation to the `required` column parameter.
* Align `Ascii` column behaviour to `Text`, by adding `min_length` and `max_length` parameter support.

### CODE REVIEW

**Comment by kdeldycke** [comment]
Ah. Just realized that there's two sets of tests: `unit` and `integration`. Unfortunately the later are not launched by Travis. Is this expected?

**Comment by GregBestland** [comment]
Unfortunately our integration tests are a little too hefty to run on travis. We have some separate internal infrastructure to run these. It is expected.

**Comment by kdeldycke** [comment]
Both unit and integration tests pass on my machine.

This PR is ready to be reviewed and merged.

**Comment by kdeldycke** [comment]
As a side note, this covers old https://github.com/cqlengine/cqlengine/pull/283 PR

**Comment by GregBestland** [comment]
We will use this PR, instead of mine. I like the way you handled the validation better.

**Comment by kdeldycke** [comment]
OK, as you wish. Do not hesitate to tell me of this PR needs some changes or improvements.

For instance, do you want me to separate the string handling logic in its own class? Like you did with the [`BaseStringColumn` class](https://github.com/datastax/python-driver/pull/628/files#diff-329bf32e861903f9ac70eca515f8069aR312)?

File: cassandra/cqlengine/columns.py [hunk 1]
```diff
@@ -335,25 +328,49 @@ def __init__(self, min_length=None, max_length=None, **kwargs):
...
                 raise ValidationError('{0} is longer than {1} characters'.format(self.column_name, self.max_length))
         if self.min_length:
-            if len(value) < self.min_length:
+            if (self.min_length and not value) or len(value) < self.min_length:
                 raise ValidationError('{0} is shorter than {1} characters'.format(self.column_name, self.min_length))
         return value
 
 
+class Ascii(Text):
+    """
```
Comment (GregBestland) [inline cassandra/cqlengine/columns.py]: We either need a separate class extending the init and validation logic, or Text needs to extend Ascii, either way is fine.

File: tests/integration/cqlengine/columns/test_validation.py [hunk 2]
```diff
@@ -337,50 +338,235 @@ def test_default_zero_fields_validate(self):
...
+
+        with self.assertRaises(ValidationError):
+            Ascii().validate(5)
+
+        with self.assertRaises(ValidationError):
+            Ascii().validate(True)
+
+    def test_unaltering_validation(self):
+        """ Test the validation step doesn't re-interpret values. """
+        self.assertEquals(Ascii().validate(''), '')
```
Comment (GregBestland) [inline tests/integration/cqlengine/columns/test_validation.py]: assertEquals needs to be swapped over to assertEqual, as assertEquals is slated for depreciation in the future.

**Comment by GregBestland** [comment]
We were also wondering what the priority level of this change is for you?  Is it blocking a critical path for you?

File: cassandra/cqlengine/columns.py (hunk 1 above)
Comment (kdeldycke) [inline cassandra/cqlengine/columns.py]: The reason I was thinking the other way around is that I'm about to add a `validation()` in `Ascii` to extend on `Text`'s, and add stricter checks on legal ASCII character range. Do you want me to implement this?

File: tests/integration/cqlengine/columns/test_validation.py (hunk 2 above)
Comment (kdeldycke) [inline tests/integration/cqlengine/columns/test_validation.py]: Good catch! Done in commit: 039149aa4377abe223ab455582132c277b3e7851

**Comment by kdeldycke** [comment]
This is not highly critical. But once this PR is merged upstream I'll be able to delete some hard-coded checks in my proprietary code, to rely exclusively on CQLengine! :sweat_smile:

File: cassandra/cqlengine/columns.py (hunk 1 above)
Comment (kdeldycke) [inline cassandra/cqlengine/columns.py]: Just added the extra validation on ASCII characters in commit https://github.com/datastax/python-driver/pull/623/commits/1a72f87554f20a0e89bb15ea5389b8d8c0e2255e .

**Comment by kdeldycke** [comment]
I addressed all comments on this PR. If you have nothing to add, this is ready for a merge.

**Comment by GregBestland** [comment]
Thank you so much for your patience and assistance. Sorry it took me so long to get back to you.

 This will be included in our next release. 👍

**Comment by kdeldycke** [comment]
Thanks @GregBestland for the merge! Really happy to contribute! :)
